pkg/apis/monitoring/v1/types.go

```diff
@@ -342,38 +343,28 @@ type WebTLSConfig struct {
 	CurvePreferences []string `json:"curvePreferences,omitempty"`
 }
 
-// WebTLSConfigError is returned by WebTLSConfig.Validate() on
-// semantically invalid configurations.
-// +k8s:openapi-gen=false
-type WebTLSConfigError struct {
-	err string
-}
-
-func (e *WebTLSConfigError) Error() string {
-	return e.err
-}
-
+// Validate returns an error if one of the WebTLSConfig fields is invalid.
+// A valid WebTLSConfig should have Cert and KeySecret fields which are not
+// zero values.
 func (c *WebTLSConfig) Validate() error {
 	if c == nil {
 		return nil
 	}
 
 	if c.ClientCA != (SecretOrConfigMap{}) {
 		if err := c.ClientCA.Validate(); err != nil {
-			msg := fmt.Sprintf("invalid web tls config: %s", err.Error())
```

Comment:
Removing the "invalid web tls config" prefix since the error is already wrapped by the caller. It also enables using the same field for Alertmanager mTLS.